### PR TITLE
GEODE-9936: Modified multi-key commands for wrong type key

### PR DESCRIPTION
**Pull request by Kris-10-0**
### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [NA] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

File: geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java [hunk 1]
```diff
@@ -108,21 +104,18 @@ public static int sdiffstore(RegionProvider regionProvider, RedisKey destination
   private static MemberSet calculateDiff(RegionProvider regionProvider, List<RedisKey> keys,
       boolean updateStats) {
     RedisSet firstSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(0), updateStats);
+    MemberSet diff = new MemberSet();
     if (firstSet.scard() == 0) {
```
Comment (dschneider-pivotal) [inline geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java:108]: Since we never store a empty RedisSet (if it is empty it just gets removed) scard will never return 0. You should delete this if block.

File: geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java [hunk 2]
```diff
@@ -108,21 +104,18 @@ public static int sdiffstore(RegionProvider regionProvider, RedisKey destination
...
-      return null;
+      return diff;
     }
-    MemberSet diff = new MemberSet(firstSet.members);
 
+    diff = new MemberSet(firstSet.members);
     for (int i = 1; i < keys.size(); i++) {
       RedisSet curSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(i), updateStats);
       if (curSet.scard() == 0) {
```
Comment (dschneider-pivotal) [inline geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java:115]: Since we never store a empty RedisSet (if it is empty it just gets removed) scard will never return 0. You should delete this if block.

File: geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java (hunk 1 above)
Comment (DonalEvans) [reply geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java:108]: I think that if there is no value associated with the key then we return a `NULL_REDIS_SET` here, which returns 0 when `scard()` is called on it, so this check is actually doing something.

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffIntegrationTest.java [hunk 3]
```diff
@@ -167,6 +167,30 @@ public void sdiffWithDifferentyKeyType_returnsWrongTypeError() {
     assertThatThrownBy(() -> jedis.sdiff("ding")).hasMessageContaining(ERROR_WRONG_TYPE);
   }
 
+  @Test
+  public void sdiff_withDifferentKeyTypeAndTwoSetKeys_returnsWrongTypeError() {
+    String diffTypeKey = "{tag1}ding";
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffIntegrationTest.java:172]: Could this (and the other variables with the same name added in this PR) be renamed to `stringKey` to avoid confusion around "diff" meaning different vs the DIFF operation?

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffIntegrationTest.java [hunk 4]
```diff
@@ -167,6 +167,30 @@ public void sdiffWithDifferentyKeyType_returnsWrongTypeError() {
     assertThatThrownBy(() -> jedis.sdiff("ding")).hasMessageContaining(ERROR_WRONG_TYPE);
   }
 
+  @Test
+  public void sdiff_withDifferentKeyTypeAndTwoSetKeys_returnsWrongTypeError() {
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffIntegrationTest.java:171]: This test name might be a little clearer as something like "sdiff_withNonSetKeyAsFirstKey_returnsWrongTypeError"

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffIntegrationTest.java [hunk 5]
```diff
@@ -167,6 +167,30 @@ public void sdiffWithDifferentyKeyType_returnsWrongTypeError() {
...
+
+    String[] members = createKeyValuesSet();
+    String secondSetKey = "{tag1}secondKey";
+    jedis.sadd(secondSetKey, members);
+    assertThatThrownBy(() -> jedis.sdiff(diffTypeKey, setKey, secondSetKey))
+        .hasMessageContaining(ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void sdiff_withTwoSetKeysAndDifferentKeyType_returnsWrongTypeError() {
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffIntegrationTest.java:183]: This test name might be a little clearer as something like "sdiff_withNonSetKeyAsThirdKey_returnsWrongTypeError"

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffStoreIntegrationTest.java [hunk 6]
```diff
@@ -66,6 +66,30 @@ public void sdiffstore_DifferentKeyType_returnsWrongTypeError() {
         .hasMessageContaining(ERROR_WRONG_TYPE);
   }
 
+  @Test
+  public void sdiffstore_withDifferentKeyTypeAndTwoSetKeys_returnsWrongTypeError() {
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffStoreIntegrationTest.java:70]: This test name might be a little clearer as something like "sdiffstore_withNonSetKeyAsFirstSourceKey_returnsWrongTypeError"

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffStoreIntegrationTest.java [hunk 7]
```diff
@@ -66,6 +66,30 @@ public void sdiffstore_DifferentKeyType_returnsWrongTypeError() {
...
+
+    String secondSetKey = "{tag1}secondKey";
+    jedis.sadd(setKey, setMembers);
+    jedis.sadd(secondSetKey, setMembers);
+    assertThatThrownBy(() -> jedis.sdiffstore(destinationKey, diffTypeKey, setKey, secondSetKey))
+        .hasMessageContaining(ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void sdiffstore_withTwoSetKeysAndDifferentKeyType_returnsWrongTypeError() {
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffStoreIntegrationTest.java:82]: This test name might be a little clearer as something like "sdiffstore_withNonSetKeyAsThirdSourceKey_returnsWrongTypeError"

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java [hunk 8]
```diff
@@ -78,6 +78,32 @@ public void shouldError_givenWrongKeyType() {
         STRING_KEY, KEY1)).hasMessage("WRONGTYPE " + RedisConstants.ERROR_WRONG_TYPE);
   }
 
+  @Test
+  public void shouldError_givenTwoSortedSetKeysAndWrongKeyType() {
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java:82]: This test name might be a little clearer as something like "shouldReturnWrongTypeError_givenNonSortedSetKeyAsThirdSourceKey"

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java [hunk 9]
```diff
@@ -78,6 +78,32 @@ public void shouldError_givenWrongKeyType() {
...
+
+    final String STRING_KEY = "{tag1}stringKey";
+    jedis.set(STRING_KEY, "value");
+
+    assertThatThrownBy(() -> jedis.zinterstore(NEW_SET, KEY1, KEY2,
+        STRING_KEY)).hasMessageContaining(RedisConstants.ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void shouldError_givenWrongKeyTypeAndTwoSortedSetKeys() {
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java:96]: This test name might be a little clearer as something like "shouldReturnWrongTypeError_givenNonSortedSetKeyAsFirstSourceKey". Also, this test case duplicates the test "shouldError_givenWrongKeyType" but is clearer than that one, so that one should probably be removed.

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java [hunk 10]
```diff
@@ -70,6 +70,44 @@ public void shouldError_givenWrongKeyType() {
...
+    scores1.put("player1", Double.NEGATIVE_INFINITY);
+    scores1.put("player2", 0D);
+    scores1.put("player3", 1D);
+    scores1.put("player4", Double.POSITIVE_INFINITY);
+    jedis.zadd(KEY1, scores1);
+
+    Map<String, Double> scores2 = makeScoreMap(10, x -> (double) (9 - x));
+    jedis.zadd(KEY2, scores2);
+
+    final String DIFF_TYPE_KEY = "{tag1}stringKey";
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java:85]: Could this be renamed to "stringKey" please?

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java [hunk 11]
```diff
@@ -70,6 +70,44 @@ public void shouldError_givenWrongKeyType() {
...
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java:104]: Could this be renamed to "stringKey" please?

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java [hunk 12]
```diff
@@ -70,6 +70,44 @@ public void shouldError_givenWrongKeyType() {
                 .hasMessage("WRONGTYPE " + RedisConstants.ERROR_WRONG_TYPE);
   }
 
+  @Test
+  public void shouldError_givenTwoSortedSetKeysAndWrongKeyType() {
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java:74]: This test name might be a little clearer as something like "shouldReturnWrongTypeError_givenNonSortedSetKeyAsThirdSourceKey"

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java [hunk 13]
```diff
@@ -70,6 +70,44 @@ public void shouldError_givenWrongKeyType() {
...
+
+    final String DIFF_TYPE_KEY = "{tag1}stringKey";
+    jedis.set(DIFF_TYPE_KEY, "value");
+
+    assertThatThrownBy(() -> jedis.zunionstore(NEW_SET, KEY1, KEY2, DIFF_TYPE_KEY))
+        .hasMessageContaining(RedisConstants.ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void shouldError_givenWrongKeyTypeAndTwoSortedSetKeys() {
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java:93]: This test name might be a little clearer as something like "shouldReturnWrongTypeError_givenNonSortedSetKeyAsFirstSourceKey". Also, this test case duplicates the test "shouldError_givenWrongKeyType" but is clearer than that one, so that one should probably be removed.

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java [hunk 14]
```diff
@@ -78,6 +78,32 @@ public void shouldError_givenWrongKeyType() {
         STRING_KEY, KEY1)).hasMessage("WRONGTYPE " + RedisConstants.ERROR_WRONG_TYPE);
   }
 
+  @Test
+  public void shouldError_givenTwoSortedSetKeysAndWrongKeyType() {
+    Map<String, Double> scores = buildMapOfMembersAndScores(1, 5);
+    Map<String, Double> nonIntersectionScores = buildMapOfMembersAndScores(6, 10);
+    jedis.zadd(KEY1, scores);
+    jedis.zadd(KEY2, nonIntersectionScores);
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java:86]: The set-up in this test can be simplified, since we don't really care what's in the sorted sets, just that they ARE sorted sets. Something like this maybe:
```
    jedis.zadd(KEY1, 1, "value1");
    jedis.zadd(KEY2, 1, "value2");
```

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java [hunk 15]
```diff
@@ -78,6 +78,32 @@ public void shouldError_givenWrongKeyType() {
...
+
+    final String STRING_KEY = "{tag1}stringKey";
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java:88]: Minor point, but all-caps variable names should be reserved for static constants rather than local variables. (See the [Google Java Style guide](https://google.github.io/styleguide/javaguide.html#s5.2-specific-identifier-names) that Geode follows). I think the reason there are some local variables with the wrong format in this file already is due to previous refactoring when they were turned from constants into local variables but not renamed.

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java [hunk 16]
```diff
@@ -78,6 +78,32 @@ public void shouldError_givenWrongKeyType() {
...
+    jedis.set(STRING_KEY, "value");
+
+    assertThatThrownBy(() -> jedis.zinterstore(NEW_SET, KEY1, KEY2,
+        STRING_KEY)).hasMessageContaining(RedisConstants.ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void shouldError_givenWrongKeyTypeAndTwoSortedSetKeys() {
+    Map<String, Double> scores = buildMapOfMembersAndScores();
+    jedis.zadd(KEY1, scores);
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java:98]: This set-up can also be simplified:
```
    jedis.zadd(KEY1, 1, "value1");
    jedis.zadd(KEY2, 1, "value2");
```

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java [hunk 17]
```diff
@@ -70,6 +70,44 @@ public void shouldError_givenWrongKeyType() {
...
+  public void shouldError_givenTwoSortedSetKeysAndWrongKeyType() {
+    Map<String, Double> scores1 = new LinkedHashMap<>();
+    scores1.put("player1", Double.NEGATIVE_INFINITY);
+    scores1.put("player2", 0D);
+    scores1.put("player3", 1D);
+    scores1.put("player4", Double.POSITIVE_INFINITY);
+    jedis.zadd(KEY1, scores1);
+
+    Map<String, Double> scores2 = makeScoreMap(10, x -> (double) (9 - x));
+    jedis.zadd(KEY2, scores2);
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java:83]: See above comments about simplifying the set-up for these tests.

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java [hunk 18]
```diff
@@ -70,6 +70,44 @@ public void shouldError_givenWrongKeyType() {
...
+  public void shouldError_givenWrongKeyTypeAndTwoSortedSetKeys() {
+    Map<String, Double> scores1 = new LinkedHashMap<>();
+    scores1.put("player1", Double.NEGATIVE_INFINITY);
+    scores1.put("player2", 0D);
+    scores1.put("player3", 1D);
+    scores1.put("player4", Double.POSITIVE_INFINITY);
+    jedis.zadd(KEY1, scores1);
+
+    Map<String, Double> scores2 = makeScoreMap(10, x -> (double) (9 - x));
+    jedis.zadd(KEY2, scores2);
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZUnionStoreIntegrationTest.java:102]: See above comments about simplifying the set-up for these tests.

File: geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java [hunk 19]
```diff
@@ -108,21 +104,18 @@ public static int sdiffstore(RegionProvider regionProvider, RedisKey destination
...
     RedisSet firstSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(0), updateStats);
+    MemberSet diff = new MemberSet();
     if (firstSet.scard() == 0) {
-      return null;
+      return diff;
     }
-    MemberSet diff = new MemberSet(firstSet.members);
 
+    diff = new MemberSet(firstSet.members);
     for (int i = 1; i < keys.size(); i++) {
```
Comment (DonalEvans) [inline geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java:113]: This early return causes incorrect behaviour if the first set key does not exist. The following test passes with native Redis but fails with geode-for-redis:
```
  @Test
  public void sdiff_withNonSetKeyAsThirdKeyAndNonExistentSetAsFirstKey_returnsWrongTypeError() {
    String stringKey = "{tag1}ding";
    jedis.set(stringKey, "dong");

    String secondSetKey = "{tag1}secondKey";
    jedis.sadd(secondSetKey, "member");
    assertThatThrownBy(() -> jedis.sdiff("emptyKey", secondSetKey, stringKey))
        .hasMessageContaining(ERROR_WRONG_TYPE);
  }
```

To fix the issue, this block should be:
```
    RedisSet firstSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(0), updateStats);

    MemberSet diff = new MemberSet(firstSet.members);
    for (int i = 1; i < keys.size(); i++) {

```

**Review by DonalEvans — CHANGES_REQUESTED** [review]

**Comment by DonalEvans** [comment]
In addition to the comments I left, this test if added to `AbstractZInterStoreIntegrationTest` also passes with native Redis but fails with geode-for-redis:
```
  @Test
  public void shouldReturnWrongTypeError_withNonSortedSetKeyAsThirdSourceKeyAndNonExistentSortedSetAsFirstSourceKey() {
    jedis.zadd(KEY2, 1, "value2");

    final String STRING_KEY = "{tag1}stringKey";
    jedis.set(STRING_KEY, "value");

    assertThatThrownBy(() -> jedis.zinterstore(NEW_SET, "{tag1}nonExistentKey", KEY2,
        STRING_KEY)).hasMessageContaining(RedisConstants.ERROR_WRONG_TYPE);
  }
```

This can be fixed by changing the for loop in the `zinterstore()` method to the following:
```
    for (ZKeyWeight keyWeight : keyWeights) {
      RedisSortedSet set =
          regionProvider.getTypedRedisData(REDIS_SORTED_SET, keyWeight.getKey(), false);

      if (set != NULL_REDIS_SORTED_SET) {
        sets.add(set);
      }
    }
```

File: geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java (hunk 1 above)
Comment (Kris-10-0) [reply geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java]: I changed it to NULL_REDIS_SET, because I thought it would make it easier to understand. Let me know if I should change it back.

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java (hunk 15 above)
Comment (Kris-10-0) [reply geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java]: Should I rename the other local variables in the class to follow that format.

Comment (DonalEvans) [reply geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java]: That would be awesome, thanks!

**Comment by Kris-10-0** [comment]
Forced pushed to rebase

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffIntegrationTest.java [hunk 20]
```diff
@@ -133,38 +133,56 @@ public void sdiffSetsNotModified_returnSetValues() {
...
+
+    String[] members = createKeyValuesSet();
+    String secondSetKey = "{tag1}secondKey";
+    jedis.sadd(secondSetKey, members);
+    assertThatThrownBy(() -> jedis.sdiff(stringKey, SET_KEY, secondSetKey))
+        .hasMessageContaining(ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void sdiff_withNonSetKeyAsThirdKey_returnsWrongTypeError() {
```
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSDiffIntegrationTest.java:177]: Could we also get tests of the behaviour when there are three keys, the first key does not exist, and the third key is not a set? This is a distinct error case that this PR fixes the behaviour for, but doesn't currently have tests for. An example for the SDIFF case would be:
```
  @Test
  public void sdiff_withNonSetKeyAsThirdKeyAndNonExistentSetAsFirstKey_returnsWrongTypeError() {
    String stringKey = "{tag1}ding";
    jedis.set(stringKey, "dong");

    String secondSetKey = "{tag1}secondKey";
    jedis.sadd(secondSetKey, "member");
    assertThatThrownBy(() -> jedis.sdiff(NON_EXISTENT_SET_KEY, secondSetKey, stringKey))
        .hasMessageContaining(ERROR_WRONG_TYPE);
  }
```
but we should probably have a test like this for all the multi-key commands that can return WRONGTYPE errors to make sure we're returning what native Redis does.

File: geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java [hunk 21]
```diff
@@ -287,22 +287,32 @@ public long zcount(SortedSetScoreRangeOptions rangeOptions) {
...
       if (set == NULL_REDIS_SORTED_SET) {
-        return sortedSetOpStoreResult(regionProvider, key, new MemberMap(0), new ScoreSet());
+        noIntersection = true;
       } else {
         sets.add(set);
       }
     }
 
+    if (noIntersection) {
+      return sortedSetOpStoreResult(regionProvider, destinationKey, new MemberMap(0),
```
Comment (DonalEvans) [inline geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java:312]: Technically unrelated to this PR, but the arguments for the `sortedSetOpStoreResult()` method should probably be renamed. `MemberMap interMembers` and `ScoreSet interScores` should probably just be `MemberMap members` and `ScoreSet scores`. If it's not too much bother, would you mind renaming them, since you're already modifying this class?

**Review by DonalEvans — CHANGES_REQUESTED** [review]
Just a few missing tests, but other than that, looks really good.

**Review by DonalEvans — APPROVED** [review]
Thanks for applying all the review comments!
